src/utilities/effects.ts

```diff
@@ -3700,11 +3700,26 @@ const effects = {
             const data = await reqHasura<SchedulingResponse>(gql.SCHEDULE, { specificationId }, user);
             const { schedule } = data;
             if (schedule != null) {
-              const { reason, status } = schedule;
+              const { datasetId, reason, status } = schedule;
 
               if (status === 'complete') {
                 schedulingStatus.set(Status.Complete);
                 incomplete = false;
+                if (datasetId != null) {
+                  const simDatasetIdData = await reqHasura<{ id: number }>(
+                    gql.SIMULATION_DATASET_ID,
+                    { datasetId },
+                    user,
+                  );
+                  const { simulation_dataset } = simDatasetIdData;
+                  if (Array.isArray(simulation_dataset) && simulation_dataset.length > 0) {
+                    simulationDatasetId.set(simulation_dataset[0].id);
```

Comment:
This should never return more than one entry in the response array, right?

Reply:
Right

Reply:
Actually do you mind just adding a small comment here that explains that the array will only ever return 0 or 1, please? Just so that the next person can immediately know what's going on rather than trying to interpret it from looking at the query.